### DISCUSSION TITLE
nx affected with test & lint

### DISCUSSION THREAD

**Discussion by danielcornock** [nrwl/nx#7046] [General]
Fairly new to the nx world, and have been playing with the affected command.

We currently only have one app, with multiple libraries.

My intention is to have nx affected only lint and test projects (including libs) that have been changed (including things that depend on them).

When I run nx affected:dep-graph, it correctly highlights in red the libraries that have been changed (and things that depend on them), but then when I actually run nx affected:test, it tests all of the libraries.

I understand for build that it would need to do it on a per-app basis, but is there a way to run all tests on only apps and libs that are directly affected by your changes?

I feel like I may be missing something obvious here

**Comment by Mellbourn** [comment]
Have you tried `nx affected --target=test`, where `test` is defined in the `project.json` of each project?
